control-plane/controllers/resources/mesh_gateway_controller_test.go

```diff
@@ -50,6 +52,16 @@ func TestMeshGatewayController_Reconcile(t *testing.T) {
 						Namespace: "consul",
 						Name:      "mesh-gateway",
 					},
+					Spec: pbmesh.MeshGateway{
+						GatewayClassName: "consul",
+						Listeners: []*pbmesh.MeshGatewayListener{
+							{
+								Name:     "wan",
+								Port:     8443,
+								Protocol: "mesh",
```

Comment:
This is currently required to be `TCP` [here](https://github.com/hashicorp/consul/blob/6a8554317502dc54fb794ce950db8678ec352c55/proto-public/pbmesh/v2beta1/mesh_gateway.proto#L29-L30). Will be explicitly `mesh` in the future; however, I'd make it TCP for now in these tests so as not to run into any surprises.